### PR TITLE
fix: Type and story fixes for Segments

### PR DESCRIPTION
**Pull request by giulioz**
### Why

The current loading times of the Segments story are pretty slow, since it's calling `useFrame` for each object. Unfortunately this is slow in r3f, since it has to update the internal zustand state multiple times. I think a better example for performance can be shown by using an array of refs and a single useFrame.

During this process I also found that some user-facing types of the component were wrong, so I corrected them.

### What

(see above)

### Checklist

- [x] Documentation updated
- [x] Storybook entry added
- [x] Ready to be merged

### CODE REVIEW

File: src/core/Segments.tsx [hunk 1]
```diff
@@ -11,24 +11,21 @@ type SegmentsProps = {
...
-type SegmentRef = React.RefObject<Segment>
-type SegmentProps = JSX.IntrinsicElements['segmentObject'] & Segment
-
 const context = React.createContext<Api>(null!)
 
-const arrColor = (color) => (color instanceof THREE.Color ? color.toArray() : color)
-const arrPos = (pos) => (pos instanceof THREE.Vector3 ? pos.toArray() : pos)
-
 const Segments = React.forwardRef<Line2, SegmentsProps>((props, forwardedRef) => {
+  React.useMemo(() => extend({ SegmentObject }), [])
```
Comment (joshuaellis) [inline src/core/Segments.tsx:27]: this doesn't feel right, why wouldn't we extend in the global space? cc @drcmda @CodyJasonBennett

File: src/core/Segments.tsx [hunk 2]
```diff
@@ -37,12 +34,12 @@ const Segments = React.forwardRef<Line2, SegmentsProps>((props, forwardedRef) =>
   const [geometry] = React.useState(() => new LineSegmentsGeometry())
   const [resolution] = React.useState(() => new THREE.Vector2(512, 512))
 
-  const [positions] = React.useState(() => Array(limit * 6).fill(0))
-  const [colors] = React.useState(() => Array(limit * 6).fill(0))
+  const [positions] = React.useState<number[]>(() => Array(limit * 6).fill(0))
```
Comment (joshuaellis) [inline src/core/Segments.tsx:37]: was this not infered?

**Review by joshuaellis — CHANGES_REQUESTED** [review]

**Comment by joshuaellis** [comment]
Few Qs but nothing majro

File: src/core/Segments.tsx (hunk 1 above)
Comment (giulioz) [reply src/core/Segments.tsx:27]: We do that also in the Instancing component

Comment (giulioz) [reply src/core/Segments.tsx:27]: https://github.com/pmndrs/drei/blob/bc6e3b0ddc5078cdd77e6385ffad0efa45dbf4d1/src/core/Instances.tsx#L39

Comment (giulioz) [reply src/core/Segments.tsx:27]: Furthermore, we do that for each instance

File: src/core/Segments.tsx (hunk 2 above)
Comment (giulioz) [reply src/core/Segments.tsx:37]: No, because .fill doesn't change the return type of the array

Comment (joshuaellis) [reply src/core/Segments.tsx:37]: til

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**